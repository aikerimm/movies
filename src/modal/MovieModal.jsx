import PropTypes from 'prop-types';
import './movieModal.css';
import React from 'react';
import MovieForm from './MovieForm.jsx';

const MovieModal = ({ onClose, modalTitle, onSubmit, movie }) => {
  return (
    <div className='overlay' onClick={onClose}>
      <div
        className='modalContainer'
        onClick={(event) => event.stopPropagation()}
      >
        <p className='closeBtn' onClick={onClose}>
          X
        </p>
        <p className='titleText fullWidth'>{modalTitle}</p>
        <MovieForm movie={movie} onMovieFormSubmit={onSubmit} />
      </div>
    </div>
  );
};

MovieModal.propTypes = {
  onClose: PropTypes.func.isRequired,
  modalTitle: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
  movie: PropTypes.object,
};

MovieModal.defaultProps = {
  showSuccessModal: false,
};

export default React.memo(MovieModal);
