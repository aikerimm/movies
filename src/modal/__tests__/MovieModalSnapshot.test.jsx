import renderer from 'react-test-renderer';
import MovieModal from '../MovieModal';

it('MovieModal should render correctly', () => {
  const movie = {id: 123};
  const tree = renderer.create(<MovieModal modalTitle='Title' onClose={jest.fn()} onSubmit={jest.fn()} movie={movie}/>).toJSON();
  expect(tree).toMatchSnapshot();
});