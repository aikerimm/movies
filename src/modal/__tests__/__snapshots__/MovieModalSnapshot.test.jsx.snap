// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieModal should render correctly 1`] = `
<div
  className="overlay"
  onClick={[MockFunction]}
>
  <div
    className="modalContainer"
    onClick={[Function]}
  >
    <p
      className="closeBtn"
      onClick={[MockFunction]}
    >
      X
    </p>
    <p
      className="titleText fullWidth"
    >
      Title
    </p>
    <form
      className="addMovieForm"
      onSubmit={[Function]}
    >
      <div
        className="formInput"
      >
        <label
          htmlFor="title"
        >
          Title
        </label>
        <input
          id="title"
          name="title"
          onChange={[Function]}
          placeholder="Movie Title"
          type="text"
        />
      </div>
      <div
        className="formInput"
      >
        <label
          htmlFor="release_date"
        >
          Release Date
        </label>
        <input
          id="release_date"
          name="release_date"
          onChange={[Function]}
          type="date"
        />
      </div>
      <div
        className="formInput"
      >
        <label
          htmlFor="movieUrl"
        >
          Movie Url
        </label>
        <input
          id="poster_path"
          name="poster_path"
          onChange={[Function]}
          placeholder="https://"
          type="url"
        />
      </div>
      <div
        className="formInput"
      >
        <label
          htmlFor="vote_average"
        >
          Rating
        </label>
        <input
          id="vote_average"
          name="vote_average"
          onChange={[Function]}
          placeholder="for ex. 9.9"
          type="text"
        />
      </div>
      <div
        className="formInput"
      >
        <label
          htmlFor="genres"
        >
          Genre
        </label>
        <select
          id="genres"
          multiple={true}
          name="genres"
          onChange={[Function]}
        >
          <option
            value="Action"
          >
            Action
          </option>
          <option
            value="Adventure"
          >
            Adventure
          </option>
          <option
            value="Animation"
          >
            Animation
          </option>
          <option
            value="Comedy"
          >
            Comedy
          </option>
          <option
            value="Crime"
          >
            Crime
          </option>
          <option
            value="Drama"
          >
            Drama
          </option>
          <option
            value="Family"
          >
            Family
          </option>
          <option
            value="Fantasy"
          >
            Fantasy
          </option>
          <option
            value="History"
          >
            History
          </option>
          <option
            value="Horror"
          >
            Horror
          </option>
          <option
            value="Music"
          >
            Music
          </option>
          <option
            value="Mystery"
          >
            Mystery
          </option>
          <option
            value="Romance"
          >
            Romance
          </option>
          <option
            value="Science Fiction"
          >
            Science Fiction
          </option>
          <option
            value="Thriller"
          >
            Thriller
          </option>
          <option
            value="War"
          >
            War
          </option>
        </select>
      </div>
      <div
        className="formInput"
      >
        <label
          htmlFor="runtime"
        >
          Runtime
        </label>
        <input
          id="runtime"
          name="runtime"
          onChange={[Function]}
          placeholder="in minutes"
          type="text"
        />
      </div>
      <div
        className="formInput overviewInput"
      >
        <label
          htmlFor="overview"
        >
          Overview
        </label>
        <textarea
          id="overview"
          name="overview"
          onChange={[Function]}
          placeholder="Movie Description"
          rows="4"
        />
      </div>
      <div
        className="btnContainer"
      >
        <input
          className="cancelBtn roundedCorners"
          onClick={[Function]}
          type="button"
          value="Reset"
        />
        <button
          className="submitBtn roundedCorners"
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</div>
`;
